internal/services/observer.go

```diff
@@ -0,0 +1,171 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"github.com/shulganew/gophermart/internal/config"
+	"github.com/shulganew/gophermart/internal/model"
+	"go.uber.org/zap"
+)
+
+// Check Acceral service every X sec
+const CheckAccrual = 1
+
+// Check Oraders in DB every X sec
+const UploadData = 3
+
+type AccrualResponce struct {
+	Order   string  `json:"order"`
+	Status  string  `json:"status"`
+	Accrual float64 `json:"accrual"`
+}
+
+type Observer struct {
+	stor ObserverUpdater
+	conf *config.Config
+
+	// map[order number]Order
+	orders map[string]model.Order
+	mu     sync.Mutex
+}
+
+type ObserverUpdater interface {
+	LoadPocessing(ctx context.Context) ([]model.Order, error)
+	UpdateStatus(ctx context.Context, order *model.Order, accrual *decimal.Decimal) error
+}
+
+func NewObserver(stor ObserverUpdater, conf *config.Config) *Observer {
+	return &Observer{stor: stor, conf: conf, orders: make(map[string]model.Order, 0)}
+}
+
+func (o *Observer) AddOreder(order *model.Order) {
```

Comment:
```suggestion
func (o *Observer) AddOrder(order *model.Order) {
```
а что если передадим nil?

Reply:
Поработал с ссылками, убрал лишние